tensorflow/core/util/sparse/sparse_tensor.h

```diff
@@ -530,6 +537,65 @@ std::vector<SparseTensor> SparseTensor::Split(const SparseTensor& input_tensor,
   return output_tensors;
 }
 
+template <typename T>
+SparseTensor SparseTensor::Slice(const SparseTensor& input_tensor,
+                                 const gtl::ArraySlice<int64> &start,
+                                 const gtl::ArraySlice<int64> &size) {
+  TensorShape output_shape = input_tensor.shape();
+
+  const int dims = input_tensor.shape().dims();
+  for (int dim = 0; dim < dims; dim++) {
+    int64 dim_size = start[dim] + size[dim] < output_shape.dim_size(dim) ? size[dim] : output_shape.dim_size(dim) - start[dim];
+    output_shape.set_dim(dim, dim_size);
+  }
+
+  auto input_indices_t = input_tensor.indices().matrix<int64>();
+  auto input_values_t = input_tensor.values().vec<T>();
+
+  int count = 0;
+  for (int i = 0; i < input_tensor.indices().dim_size(0); i++) {
+    bool hit = true;
+    for (int dim = 0; dim < dims; dim++) {
+      if (!(start[dim] <= input_indices_t(i, dim) && input_indices_t(i, dim) < start[dim] + size[dim])) {
```

Comment:
are you making assumptions about the sorting of the SparseTensor?  Do you need to run ValidateIndices first?